git-clonefork

```diff
@@ -14,15 +14,42 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if [[ -z $1 ]]; then
+    echo "Please provide a fork url"
+    echo "usage: git clonefork <fork_url>"
+    exit 2
+fi
+
 GIT_ORIGIN=$1
+# get the name of the directory that will be created when we clone
 REPO_DIRECTORY=$(basename $GIT_ORIGIN .git)
+# reformat the repo url into a github api request
 GITHUB_API_URL=$(echo $GIT_ORIGIN | sed -r 's/.*(\@|\/\/)(.*)(\:|\/)([^:\/]*)\/([^\/\.]*)\.git/https:\/\/api.github.com\/repos\/\4\/\5/')
+
+# make the api request and validate we get a good response
+# -f makes curl fail silently, so REPO_DATA will be null on failure
+REPO_DATA=$(curl -fs $GITHUB_API_URL)
+if [[ -z $REPO_DATA ]]; then
+    echo "Please provide a valid fork url"
+    echo "usage: git clonefork <fork_url>"
+    exit 2
+fi
+
 if grep -Eiq "^https?://" <<<$GIT_ORIGIN; then
-	UPSTREAM_ORIGIN=$(curl -s $GITHUB_API_URL | jq -r .parent.clone_url)
+    # we aren't cloning with ssh, use the default parent clone_url
+	UPSTREAM_ORIGIN=$(echo $REPO_DATA | jq -r .parent.clone_url)
+else
+    # we want to clone with ssh
+	UPSTREAM_ORIGIN=$(echo $REPO_DATA | jq -r .parent.ssh_url)
+fi
+
+if ! git clone $GIT_ORIGIN 2>/dev/null
+then
+    echo "Failed to clone the repository. Please use a valid url pointing to your fork"
```

Comment:
if the repo exists in the current directory, the error message should be different.
![CleanShot 2023-07-23 at 01 36 50@2x](https://github.com/Cali0707/git-utils/assets/36619969/d1ea1990-20c2-4dc7-a7d9-d965dd87f88c)

Currently. 
![CleanShot 2023-07-23 at 01 33 18@2x](https://github.com/Cali0707/git-utils/assets/36619969/0ef00458-f918-4417-8eb4-5d7363bb9b5f)

Reply:
@Leo6Leo just clarifying, in this scenario the repo exists in the current directory before rnning `git clonefork <url>`, and then you run `git clonefork <url>`?